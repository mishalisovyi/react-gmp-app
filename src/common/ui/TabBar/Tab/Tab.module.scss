@use '../../../../mixins';
@use '../../../../colors';

.Tab {
  @include mixins.typography-button-small;

  display: inline-block;
  padding: 20px 0;
  color: colors.$white;
  z-index: 1;
  cursor: pointer;

  &--active, &:hover {
    border-bottom: solid 4px colors.$festive-fennec;
  }
}
