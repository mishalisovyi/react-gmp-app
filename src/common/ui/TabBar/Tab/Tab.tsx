import React, { memo, useMemo } from 'react';
import { EnumerableComponentProps } from 'common/interfaces';
import styles from './Tab.module.scss';

interface TabProps extends EnumerableComponentProps {
  label: string;
  value: string;
  isActive: boolean;
  index: number;
  onClick: (tabIndex: number, tabLabel: string) => void;
}

function TabComponent({
  index,
  label,
  value,
  isActive,
  onClick,
}: TabProps) {
  const activeTabClass = useMemo(() => {
    return isActive ? styles['Tab--active'] : '';
  }, [isActive]);

  const handleTabClick = () => {
    onClick(index, value);
  };

  return <div className={`${styles['Tab']} ${activeTabClass}`} onClick={handleTabClick}>{label}</div>;
}

export const Tab = memo(TabComponent);
