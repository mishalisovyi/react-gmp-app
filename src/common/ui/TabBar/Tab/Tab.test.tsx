import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Tab } from './Tab';

const tabIndex = 3;
const tabValue = 'tab_value';
const tabLabel = 'Tab 4';

const tabClickedHandler = jest.fn();

it('should render tab', () => {
  const tree = renderer.create(<Tab label={tabLabel} value={tabValue} index={tabIndex} key={9784934} onClick={tabClickedHandler} isActive />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should handle tab click', async () => {
  render(<Tab label={tabLabel} value={tabValue} index={tabIndex} key={9784934} onClick={tabClickedHandler} isActive />);

  await userEvent.click(screen.getByText(tabLabel));

  expect(tabClickedHandler).toHaveBeenCalledWith(tabIndex, tabValue);
});
