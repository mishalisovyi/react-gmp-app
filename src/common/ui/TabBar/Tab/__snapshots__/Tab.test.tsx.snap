// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render tab 1`] = `
<div
  className="undefined undefined"
  onClick={[Function]}
>
  Tab 4
</div>
`;
